Packages/vcs/Lib/vtk_ui/text.py

```diff
@@ -17,11 +19,118 @@ def __set_font(font, text_props):
         text_props.SetFontFamily(VTK_FONT_FILE)
         text_props.SetFontFile(font)
 
+
+def lum_normalize(component):
+    if component <= .03928:
+        return component / 12.92
+    else:
+        return ((component + .055) / 1.055) ** 2.4
+
+
+def luminance(color):
+    r, g, b = [lum_normalize(c) for c in color]
+    lum = .2126 * r + .7152 * g + .0722 * b
+    return lum
+
+
+def contrast_ratio(fg, bg):
+    lum_fg = luminance(fg)
```

Comment:
These three functions are defined by the W3C for determining contrast ratios for accessibility; I used them to help me generate contrasting colors.